### DISCUSSION TITLE
Reuse redux data for calls between client side navigation using getServerSideProps

### DISCUSSION THREAD

**Discussion by gurmeetsinghmalhotra** [vercel/next.js#14437] [Help]
I have been using nextjs since a long time and recently updated to the latest version. And there is one thing that I cannot get my head around any help will be highly appreciated

I am using redux saga with nextjs and using getServerSideProps

Consider I have 2 pages 
Page A
Page B

Page A - uses data from api/a
Page B - uses data from api/a and api/b

**Scenarion 1 -**
Now if someone lands on page A the data of api/a is stored in redux and if someone navigates to page B from client itself - data present in redux can be used and an hit to api/b can be made to get remaining data on page B.

This is handled by rootReducer

```
import { combineReducers } from "redux";

import { HYDRATE } from "next-redux-wrapper";
import faqCategoriesReducer from "./faqCategories/faqCategories.reducer";
import categoryFaqsReducer from "./categoryFaqs/categoryFaqs.reducers";

const combinedReducer = combineReducers({
  faqCategories: faqCategoriesReducer,
  categoryFaqs: categoryFaqsReducer,
});

const rootReducer = (state, action) => {
  if (action.type === HYDRATE) {
    const nextState = {
      ...state, // use previous state
      ...action.payload, // apply delta from hydration
    };
    if (state.faqCategories && state.faqCategories.categories.length > 0)
      nextState.faqCategories = state.faqCategories; // preserve categories on client side navigation
    return nextState;
  } else {
    return combinedReducer(state, action);
  }
};

export default rootReducer;
```

Note the check "state.faqCategories" so if data from previous page A is present in redux it is maintained even for page B use.

**Scenarion 2 -** 
Now if someone lands directly on page B we have to hit both api/a and api/b to get the data and store in redux.

As below 

getServerSideProps of page B

```
export const getServerSideProps = wrapper.getServerSideProps(
  async ({ store, req, res, query, params, ...etc }) => {
    const categoryId = params.categorySlug;
    store.dispatch(requestApiAData())
    store.dispatch(requestApiBData());
    store.dispatch(END);
    await store.sagaTask.toPromise();
  }
);
```

This also works as expected.

Now, as you can see that we are making the call - 
store.dispatch(requestApiAData()) inside getServerSideProps. This hit will be sent even when the user lands directly on page B or navigates to page B from page A.

**Issue -**
How can I avoid the call of store.dispatch(requestApiAData()) when user navigates to page B from page A. As in this case data of api/a is already in the redux store.

**Solutions I have tried -**
Tried to use suggest in saga but it does not contains the data from previous page given the fact it is running on server.

**Comment by gurmeetsinghmalhotra** [comment]
@jamesmosier @timneutkens Can you please help me out?

**Reply by Nsquik** [reply]
Have u figured that out? : )

**Reply by gurmeetsinghmalhotra** [reply]
Can you try using store.getState()?

As far as I can remember I switched back to _getInitialProps_ from _getServerSideProps_ as _getInitialProps_ is still supported and we can access existing data if present using store.getState()

Do let me know if you need any other doubt or need any help
